lib/src/paint/light_paint_rect.dart

```diff
@@ -103,16 +104,23 @@ class LightPaintRect extends CustomPainter {
 
     double h = maxSize * (1 - progress) + target.size.height + offset;
 
+    path = radius > 0
+        ? _drawRRectHole(size, x, y, w, h, radius)
+        : _drawRectHole(size, x, y, w, h);
+
     canvas.drawPath(
-      radius > 0
-          ? _drawRRectHole(size, x, y, w, h, radius)
-          : _drawRectHole(size, x, y, w, h),
+      path!,
       Paint()
         ..style = PaintingStyle.fill
         ..color = colorShadow.withOpacity(opacityShadow),
     );
   }
 
+  @override
+  bool hitTest(Offset position) {
+    return path?.contains(position) ?? false;
+  }
```

Comment:
This is the part that was hard to find. Before when you painted the container with the hole and tapped on the hole it was recognized as a tap, when it was not really part of the painter, when you tap on the hole it is not recognized as a tap on the container.